src/main/resources/assets/carpet/scripts/draw_beta.sc

```diff
@@ -42,26 +52,21 @@ _block_matches(existing, block_predicate) ->
     (!tag || tag_matches(block_data(existing), tag))
 );
 
-set_block(x, y, z, block, replacement)-> (
+set_block(x, y, z, block, replacement, cpos)-> (
     existing = block(x, y, z);
-    if(block != existing && (replacement && _block_matches(existing, replacement) ),
-        without_updates( if (set(existing,block),
-            global_affected += [x, y, z];
-        ));
+    if(block != existing && (!replacement || _block_matches(existing, replacement) ),
+        global_positions:1 += bool(set(existing,block));
     );
+    global_positions:0+=[x, y ,z]-cpos//so it's relative to centre pos, and we can draw from cache more smartly
 );
 
-global_affected = [];
+global_positions = [[],0];
```

Comment:
why do you worry about checking list length?

Reply:
cos the number indicates no. of successes, list indicates all blocks in that shape.